providers/configure.rb

```diff
@@ -19,10 +19,19 @@
 #
 
 action :run do
+  sysconfig
   configure
   new_resource.updated_by_last_action(true)
 end
 
+def sysconfig
+  hugepage_file = "/sys/kernel/mm/transparent_hugepage/enabled"
+  file hugepage_file do
+    content 'never'
+    only_if { ::File.exists?(hugepage_file) }
```

Comment:
Can you add another check to verify the contents of the file? 
`&& File.read(hugepage_file).chomp == 'never'`

That way it gets set back to never if it gets changed accidently